### PR TITLE
Win fix

### PR DESCRIPTION
**Pull request by willmcgugan**
## Type of changes

- [X] Bug fix
- [ ] New feature
- [ ] Documentation / docstrings
- [X] Tests
- [ ] Other

## Checklist

- [X] I've run the latest [black](https://github.com/ambv/black) with default args on new code.
- [X] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [X] I've added tests for new code.
- [X] I accept that @willmcgugan may be pedantic in the code review.

## Description

Fixes https://github.com/PyFilesystem/pyfilesystem2/issues/256 and a number of other Windows issues. Now that we have Windows set up in appveyor, we can keep on top of Windows support

### CODE REVIEW

File: tests/test_encoding.py [hunk 1]
```diff
@@ -12,43 +12,48 @@
...
-            dirlist = test_fs.listdir("/")
-            self.assertEqual(dirlist, [self.TEST_FILENAME_UNICODE])
-            self.assertEqual(test_fs.readtext(dirlist[0]), "baz")
-
-    def test_scandir(self):
-        with OSFS(self.dir_path) as test_fs:
-            for info in test_fs.scandir("/"):
-                self.assertIsInstance(info.name, six.text_type)
-                self.assertEqual(info.name, self.TEST_FILENAME_UNICODE)
+if platform.system() != "Windows":
```
Comment (althonos) [inline tests/test_encoding.py]: Why not wrap in two `unittest.skipIf` decorators with two different conditions ?

Comment (willmcgugan) [reply tests/test_encoding.py]: It was breaking at the module level. Fsencode uses 'strict' error handler and was throwing an exception on the classvar.

Comment (althonos) [reply tests/test_encoding.py]: @willmcgugan : my bad !

**Comment by willmcgugan** [comment]
No worries! I've got Windows tests running with appveyor now, so we can catch these things in future.

File: tests/test_encoding.py (hunk 1 above)
Comment (willmcgugan) [reply tests/test_encoding.py]: No worries! I've got Windows tests running with appveyor now, so we can catch these things in future.

**Pull request merged**